### PR TITLE
fix(FileUploaderButton): allow more buttonKind options in propTypes

### PR DESCRIPTION
**Pull request by emyarod**
Closes carbon-design-system/carbon-components-react#1034

FileUploaderButton propTypes currently only expects `buttonKind` to either be `primary` or `secondary`, but the underlying `Button` component can accept additional `buttonKind` options

#### Changelog

**Changed**

* expected propTypes for `<FileUploaderButton>`'s `buttonKind` prop

### CODE REVIEW

**Comment by emyarod** [comment]
it looks like these propTypes are repeated in `Button` and `ModalWrapper` too. If moving towards separate modules for propTypes is a direction we want to explore, I would suggest consolidating them all in a `types.js` file. What do you think?

**Comment by asudoh** [comment]
Good point @emyarod! I think we need some careful thoughts put in to move all `propTypes` out to separate modules, I think the list of `oneOf` that are shared among multiple components can be extracted out.

**Comment by emyarod** [comment]
if we want to extract the list of `oneOf` propTypes I can begin with a `types.js` that contains just that for now and we can gradually add to it. how does that sound @asudoh?

**Comment by asudoh** [comment]
Sounds good @emyarod - Just in case I was not clear enough, let's limit it with only shared ones. Thanks!

**Comment by emyarod** [comment]
@asudoh I added the shared propTypes to a separate module, let me know what you think

File: src/prop-types/types.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const types = {
+  Button: {
```
[the remark this answers is not in the document]
Comment (emyarod) [reply src/prop-types/types.js]: Is the latest change more in line with what you are looking for?

**Pull request merged**